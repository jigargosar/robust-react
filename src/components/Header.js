import {AppBar, Toolbar, Typography, withStyles} from 'material-ui'
import {h} from '../hyper-script'
import S from '../sanctuary'
import connect2 from './connect2'
import Search from './Search'

const styles = {
  flex: {
    flex: 1,
  },
}

const Header = ({classes}) => {
  return h(AppBar, {position: 'static'}, [
    h(Toolbar, [
      h(
        Typography,
        {
          className: classes.flex,
          variant: 'title',
          color: 'inherit',
        },
        'Robust React App',
      ),
      h(Search),
    ]),
  ])
}

export default S.pipe([withStyles(styles), connect2({})], Header)
