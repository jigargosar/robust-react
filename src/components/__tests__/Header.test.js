import React from 'react'
import jssRender from '../../__tests__/helpers/jss-render'
import Header from '../Header'

it('renders without crashing', () => {
  const {container} = jssRender(<Header />)
  expect(container.firstChild).toMatchSnapshot()
})
