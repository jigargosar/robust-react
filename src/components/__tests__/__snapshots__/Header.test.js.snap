// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<header
  class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
>
  <div
    class="MuiToolbar-root MuiToolbar-gutters"
  >
    <h2
      class="MuiTypography-root MuiTypography-title MuiTypography-colorInherit Header-flex"
    >
      Robust React App
    </h2>
    <div
      class="MuiFormControl-root"
    >
      <div
        class="MuiInput-root Search-root MuiInput-formControl"
      >
        <input
          aria-invalid="false"
          aria-required="false"
          class="MuiInput-input Search-input MuiInput-inputType MuiInput-inputTypeSearch"
          placeholder="Search"
          type="search"
          value=""
        />
      </div>
    </div>
  </div>
</header>
`;
